src/lms-ds-loader/edfi_lms_ds_loader/Sql_lms_operations.py

```diff
@@ -188,33 +180,10 @@ def insert_new_records_to_production_for_user_relation(
             [f"\n    stg.{c}" for c in columns if c != "LMSUserSourceSystemIdentifier"]
         )
 
-        statement = f"""
-INSERT INTO
-    lms.{table}
-(
-    LMSUserIdentifier,{insert_columns}
-)
-SELECT
-    LMSUser.LMSUserIdentifier,{select_columns}
-FROM
-    lms.stg_{table} as stg
-INNER JOIN
-    lms.LMSUser
-ON
-    stg.LMSUserSourceSystemIdentifier = LMSUser.SourceSystemIdentifier
-AND
-    stg.SourceSystem = LMSUser.SourceSystem
-WHERE NOT EXISTS (
-  SELECT
-    1
-  FROM
-    lms.{table}
-  WHERE
-    SourceSystemIdentifier = stg.SourceSystemIdentifier
-  AND
-    SourceSystem = stg.SourceSystem
-)
-"""
+        statement = ""
+        if self.engine == DbEngine.MSSQL:
+            statement = MS_builder.insert_new_records_to_production_for_user_relation(
+                table, insert_columns, select_columns)
```

Comment:
Again, I think this would work on both or could be rewritten to work on both.

Reply:
After reading the entire PR, I'm fine with having two separate files even if when some of the SQL could be reused.

Reply:
I guess some of them work for both, and yes, we could call the methods of the MS_builder when engine == MSSQL or POSTGRES and the SQL applies, the main idea that I had, was to make it somehow extensible in case we needed new statements for some other engine in the future.